simperby/src/lib.rs

```diff
@@ -86,9 +92,81 @@ pub struct ConsensusVoteItem {
 #[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
 pub struct GenesisInfo {
     pub header: BlockHeader,
+    pub genesis_signature: FinalizationProof,
     pub chain_name: String,
 }
 
+impl GenesisInfo {
+    fn create_genesis_block(&self) -> Block {
+        Block {
+            header: self.header.clone(),
+            transactions: vec![],
+        }
+    }
+}
+
+/// A set of operations that may update the Simperby node state.
+///
+/// `ProposeBlock`, `SubmitConsensusVote` are explciitly performed through the `SimperbyApi` trait, and the others
+/// are done implicitly by the background task which is triggered by incoming p2p network messages.
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
+pub enum SimperbyOperation {
```

Comment:
IMO the term Operation and the use of imperative style is a little confusing for this debugging scaffolding.
How about `SimperbyOperation` -> `SimperbyLog`, `ProposeBlock` -> `BlockProposed`, and etc?

Reply:
1. `SimperbyLog` already exists.
2. I will unify the expression as if the node does something on the state. (`Receive` will be `Report`)
Ok with this?